src/Attribute/AutowireCollection.php

```diff
@@ -44,6 +46,7 @@ public function __construct(
         private readonly ?string $collection = null,
         private readonly ?string $database = null,
         ?string $client = null,
+        private readonly string|DocumentCodec|null $codec = null,
```

Comment:
How would you pass a `DocumentCodec` instance?

Reply:
I added an example, but it's rather straightforward:
```php
use MongoDB\Bundle\Attribute\AutowireCollection;
use MongoDB\Collection;
use MongoDB\Driver\ReadPreference;

class MyService
{
    public function __construct(
        #[AutowireCollection(codec: Codec::class, readPreference: new ReadPreference('secondary'))]
        private Collection $myCollection,
    ) {}
}
```